app/components/candidate_interface/previous_applications_component.rb

```diff
@@ -0,0 +1,77 @@
+module CandidateInterface
+  class PreviousApplicationsComponent < ViewComponent::Base
+    include ViewHelper
+
+    def initialize(application_form:)
+      @application_form = application_form
+    end
+
+    def render?
+      application_choices.present?
+    end
+
+    def application_choices
+      (previous_application_choices + eligible_current_application_choices)
+        .compact
+        .sort_by(&:id)
+    end
+
+    def provider_name_for(application_choice)
+      application_choice
+        .course
+        .provider
+        .name
+    end
+
+    def course_name_and_code_for(application_choice)
+      name = application_choice.course.name
+      code = application_choice.course.code
+
+      "#{name} (#{code})"
+    end
+
+    def view_application_url_for(application_choice)
+      govuk_link_to('View application', candidate_interface_review_previous_application_path(application_choice.application_form.id))
```

Comment:
We should probably add some visually-hidden text to the link text here, for screen reader users. Maybe something like:

```
<a href="#">View application<span class="govuk-visually-hidden"> for <%= application_choice.course.name %> at <%= application_choice.course.provider.name %></span></a>
```

Reply:
👍